### PR TITLE
[12.0][MIG] hr_timesheet

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Migration scripts for `hr_timesheet` module.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_timesheet/migrations/12.0.1.0/openupgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+NEW ir.model.access: hr_timesheet.access_account_analytic_account_user
+NEW ir.model.access: hr_timesheet.access_account_analytic_line_project
+NEW ir.model.access: hr_timesheet.access_uom_uom_hr_timesheet
+DEL ir.model.access: hr_timesheet.access_product_product_user
+DEL ir.model.access: hr_timesheet.access_product_template_timesheet_user
+DEL ir.model.access: hr_timesheet.access_product_uom_hr_timesheet
+NEW ir.ui.view: hr_timesheet.account_analytic_account_view_form_inherit
+NEW ir.ui.view: hr_timesheet.assets_backend
+DEL ir.ui.view: hr_timesheet.timesheet_view_calendar
+NEW uom.uom: uom.product_uom_day (noupdate)
```
Comment (pedrobaeza) [inline addons/hr_timesheet/migrations/12.0.1.0/openupgrade_analysis_work.txt]: These 2 updates are curious: they come from noupdate records in module `uom`, but here they are defined in XML as noupdate=0. On runbot, I have seen that the field is filled when you install `hr_timesheet`, no matter the original noupdate (I didn't know this). I suppose in this case it's the same, but I would like your confirmation, and this deserves at least a comment like:
```
NOTHING TO DO: Field `timesheet_widget` updated in noupdate records. Handled by standard update process.

File: odoo/openupgrade/doc/source/modules110-120.rst [hunk 2]
```diff
@@ -447,7 +447,7 @@ missing in the new release are marked with |del|.
 +--------------------------------------------+-------------------------------------------------+
 |project                                     | Done                                            |
 +--------------------------------------------+-------------------------------------------------+
-|project_timesheet_holidays                  |                                                 |
+|project_timesheet_holidays                  | Nothing to do                                   |
```
Comment (pedrobaeza) [inline odoo/openupgrade/doc/source/modules110-120.rst:450]: There's an openupgrade_analysis.txt file for this one.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: odoo/openupgrade/doc/source/modules110-120.rst (hunk 2 above)
Comment (MiquelRForgeFlow) [reply odoo/openupgrade/doc/source/modules110-120.rst:450]: Yes. And? It's a nothing to do.

Comment (pedrobaeza) [reply odoo/openupgrade/doc/source/modules110-120.rst:450]: You need to create the `openupgrade_analysis_work.txt` with the proper NOTHING TO DO.

Comment (MiquelRForgeFlow) [reply odoo/openupgrade/doc/source/modules110-120.rst:450]: but it's already created :S

Comment (pedrobaeza) [reply odoo/openupgrade/doc/source/modules110-120.rst:450]: I didn't see it... have you pushed it now? Any way, if so, we can continue when CIs finishes

File: addons/hr_timesheet/migrations/12.0.1.0/openupgrade_analysis_work.txt (hunk 1 above)
Comment (pedrobaeza) [reply addons/hr_timesheet/migrations/12.0.1.0/openupgrade_analysis_work.txt]: Can you confirm this?

File: odoo/openupgrade/doc/source/modules110-120.rst (hunk 2 above)
Comment (MiquelRForgeFlow) [reply odoo/openupgrade/doc/source/modules110-120.rst:450]: click on the 3rd commit and you should see it :S

Comment (pedrobaeza) [reply odoo/openupgrade/doc/source/modules110-120.rst:450]: OK, sorry for the noise.

File: addons/hr_timesheet/migrations/12.0.1.0/openupgrade_analysis_work.txt (hunk 1 above)
Comment (MiquelRForgeFlow) [reply addons/hr_timesheet/migrations/12.0.1.0/openupgrade_analysis_work.txt]: Confirmed, just by checking a v12 database.

Comment (pedrobaeza) [reply addons/hr_timesheet/migrations/12.0.1.0/openupgrade_analysis_work.txt]: OK, thanks

Comment (MiquelRForgeFlow) [reply addons/hr_timesheet/migrations/12.0.1.0/openupgrade_analysis_work.txt]: Wait. I checked now an old oca database migrated and the timesheet_widget was not filled. I will fully recheck properly.

Comment (MiquelRForgeFlow) [reply addons/hr_timesheet/migrations/12.0.1.0/openupgrade_analysis_work.txt]: As it doesn't work, I put them in the noupdate_changes.xml

**Comment by MiquelRForgeFlow** [comment]
ready

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**